src/utils/activity-object-helper.ts

```diff
@@ -14,25 +18,30 @@ export class ActivityObjectHelper {
   public static extractActorName(
     object: ActivityObject | Link | URL
   ): string | undefined {
-    if ((object as ActivityObject).nameMap) {
+    if (ActivityObjectHelper.hasProperty(object, "nameMap")) {
       const lang = navigator.language.substr(0, 2);
-      const activityObject = (object as ActivityObject);
+      const activityObject = object as ActivityObject;
 
-      if (lang in activityObject.nameMap) {
+      if (activityObject.nameMap && lang in activityObject.nameMap) {
         return activityObject.nameMap[lang];
       }
-    }
-
-    if ((object as ActivityObject).name) {
-      return (object as ActivityObject).name;
-    }
-
-    if ((object as Actor).preferredUsername) {
+    } else if (
+      ActivityObjectHelper.hasProperty(object, "name") &&
+      (object as Actor).name
+    ) {
+      return (object as Actor).name;
+    } else if (
+      ActivityObjectHelper.hasProperty(object, "preferredUsername") &&
```

Comment:
I removed this `hasProperty` thing as I somewhere read it's not necessary and the second condition is enough `(object as Actor).preferredUsername` to check if the property is present.